File: Accenture/12.RotateArray/code.cpp
#include <iostream>
#include <vector>
using namespace std;
void rotate(vector<int>& nums, int k) {
    int n = nums.size();
    vector<int> tmp(n);
    for(int i = 0; i < n; i++)
        tmp[(i + k) % n] = nums[i];
    nums = tmp;
}
int main() {
    int size, k;
    cin >> size;
    vector<int> nums(size);
    for(int i = 0; i < size; i++) {
        cin >> nums[i];
    }
    cin >> k;
    rotate(nums, k);
    for(int i = 0; i < size; i++) {
        cout << nums[i] << " ";
    }
}

File: Accenture/12.RotateArray/code.py
def rotate_array(nums, k):
    n = len(nums)
    tmp = [0] * n
    for i in range(n):
        tmp[(i + k) % n] = nums[i]
    return tmp
size = int(input())
nums = list(map(int, input().split()))
k = int(input())
rotated_array = rotate_array(nums, k)
print(*rotated_array)
